src/dbnode/encoding/m3tsz/timestamp_encoder.go

```diff
@@ -161,7 +161,7 @@ func (enc *TimestampEncoder) writeAnnotation(stream encoding.OStream, ant ts.Ann
 
 	stream.WriteBytes(buf[:annotationLength])
 	stream.WriteBytes(ant)
-	enc.PrevAnnotation = ant
+	enc.PrevAnnotation = append(enc.PrevAnnotation[:0], ant...)
```

Comment:
nit: split into two lines for clarity that this is being reset, i.e.

```
enc.PrevAnnotation = enc.PrevAnnotation[:0]
enc.PrevAnnotation = append(enc.PrevAnnotation, ant...)
```

Reply:
👍